### PR TITLE
Code group details page

### PR DESCRIPTION
**Pull request by izuchukwumba**
Created the code group details frontend page to enable users:
1. View general code group info
2. View file list
3. View group member list
4. Update group details (group name and image)
5. Add more group members
6. Create new files
7. Delete existing files

Video Demo: https://www.loom.com/share/bd94c556d02b42749e183e922883fe0a?sid=4e46472e-e475-4cc8-aa41-6c108c011dac
Screenshot 2024-07-06 at 3 47 17 PM

### CODE REVIEW

File: frontend/src/workStation/CodeGroup.jsx [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    isOpen: isCreateNewFileModalOpen,
+    onOpen: onCreateNewFileModalOpen,
+    onClose: onCreateNewFileModalClose,
+  } = useDisclosure();
+
+  const { groupId } = useParams();
+  const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+  const token = localStorage.getItem("jwt");
+  const navigate = useNavigate();
+  const options = {
```
Comment (azhiah) [inline frontend/src/workStation/CodeGroup.jsx:50]: Nice job making this a variable so you don't run into the authentication issue you were having again

File: frontend/src/workStation/CodeGroup.jsx [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+      setNewGroupImageUrl("");
+      setNewMembers([]);
+      fetchGroupDetails();
+      onUpdateGroupModalClose();
+      onAddMembersModalClose();
+    }
+  };
+
+  function handleAddNewMemberToList() {
+    if (newMemberInput) {
```
Comment (azhiah) [inline frontend/src/workStation/CodeGroup.jsx]: You could add checking that the member is not already in the `newMembers` list

File: frontend/src/workStation/CodeGroup.jsx [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+
+  const handleCreateNewFile = async () => {
+    try {
+      const response = await axios.post(
+        `${BACKEND_URL}/group/${groupId}/create-new-file`,
+        { newFileName },
+        options
+      );
+    } catch (error) {
+      setError("Error Creating Group. Try again");
```
Comment (azhiah) [inline frontend/src/workStation/CodeGroup.jsx]: This looks like the wrong error message

File: frontend/src/workStation/CodeGroup.jsx [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+            <ul>
+              {allFiles?.length > 0
+                ? allFiles.map((file, index) => {
+                    return (
+                      <li key={index}>
+                        <span onClick={() => handleOpenFile(file.id)}>
+                          {file.fileName} &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+                        </span>
+                        <span onClick={() => handleDeleteFile(file.id)}>
+                          delete file
```
Comment (azhiah) [inline frontend/src/workStation/CodeGroup.jsx]: nit: You should have consistent capitalization. In some places you use sentence case and in others you use no capitalization. I would recommend sticking with sentence case.

**Review by azhiah — COMMENTED** [review]
Looks good to me for the most part, just some small comments. Also since there was functionality implemented here that can't be shown in a screenshot, it would have been helpful to also have a video demonstration.

File: frontend/src/workStation/CodeGroup.jsx (hunk 1 above)
Comment (izuchukwumba) [reply frontend/src/workStation/CodeGroup.jsx:50]: Thanks!

File: frontend/src/workStation/CodeGroup.jsx (hunk 3 above)
Comment (izuchukwumba) [reply frontend/src/workStation/CodeGroup.jsx]: Thanks. Fixed here: https://github.com/izuchukwumba/peered/commit/a960d993a9510bf0b293543fdb359403c740e67f

File: frontend/src/workStation/CodeGroup.jsx (hunk 4 above)
Comment (izuchukwumba) [reply frontend/src/workStation/CodeGroup.jsx]: Thanks. Fixed here: https://github.com/izuchukwumba/peered/commit/a960d993a9510bf0b293543fdb359403c740e67f

File: frontend/src/workStation/CodeGroup.jsx (hunk 2 above)
Comment (izuchukwumba) [reply frontend/src/workStation/CodeGroup.jsx]: I have fixed this in a new commit: https://github.com/izuchukwumba/peered/commit/a960d993a9510bf0b293543fdb359403c740e67f

**Comment by izuchukwumba** [comment]
I have added the video demo @azhiah

**Review by azhiah — APPROVED** [review]

**Pull request merged**